cranelift/filetests/filetests/runtests/rotl.clif

```diff
@@ -0,0 +1,243 @@
+    return v2
+}
+; run: %rotl_i16_i32(0xe000, 0) == 0xe000
+; run: %rotl_i16_i32(0xe000, 1) == 0xc001
+; run: %rotl_i16_i32(0xef0f, 0) == 0xef0f
+; run: %rotl_i16_i32(0xef0f, 4) == 0xf0fe
+; run: %rotl_i16_i32(0xe004, 64) == 0xe004
+; run: %rotl_i16_i32(0xe004, 65) == 0xc009
+; run: %rotl_i16_i32(0xe004, 66) == 0x8013
+; run: %rotl_i16_i32(0xe004, 257) == 0xc009
+
+function %rotl_i16_i16(i16, i16) -> i16 {
+block0(v0: i16, v1: i16):
+    v2 = rotl.i16 v0, v1
+    return v2
+}
+; run: %rotl_i16_i16(0xe000, 0) == 0xe000
+; run: %rotl_i16_i16(0xe000, 1) == 0xc001
+; run: %rotl_i16_i16(0xef0f, 0) == 0xef0f
+; run: %rotl_i16_i16(0xef0f, 4) == 0xf0fe
+; run: %rotl_i16_i16(0xe004, 64) == 0xe004
+; run: %rotl_i16_i16(0xe004, 65) == 0xc009
+; run: %rotl_i16_i16(0xe004, 66) == 0x8013
+; run: %rotl_i16_i16(0xe004, 257) == 0xc009
+
+function %rotl_i16_i8(i16, i8) -> i16 {
+block0(v0: i16, v1: i8):
+    v2 = rotl.i16 v0, v1
+    return v2
+}
+; run: %rotl_i16_i8(0xe000, 0) == 0xe000
+; run: %rotl_i16_i8(0xe000, 1) == 0xc001
+; run: %rotl_i16_i8(0xef0f, 0) == 0xef0f
+; run: %rotl_i16_i8(0xef0f, 4) == 0xf0fe
+; run: %rotl_i16_i8(0xe004, 64) == 0xe004
+; run: %rotl_i16_i8(0xe004, 65) == 0xc009
+; run: %rotl_i16_i8(0xe004, 66) == 0x8013
+
+
+function %rotl_i8_i64(i8, i64) -> i8 {
+block0(v0: i8, v1: i64):
+    v2 = rotl.i8 v0, v1
+    return v2
+}
+; run: %rotl_i8_i64(0xe0, 0) == 0xe0
+; run: %rotl_i8_i64(0xe0, 1) == 0xc1
+; run: %rotl_i8_i64(0xef, 0) == 0xef
+; run: %rotl_i8_i64(0xef, 4) == 0xfe
+; run: %rotl_i8_i64(0xe4, 64) == 0xe4
+; run: %rotl_i8_i64(0xe4, 65) == 0xc9
+; run: %rotl_i8_i64(0xe4, 66) == 0x93
+; run: %rotl_i8_i64(0xe4, 257) == 0xc9
+
+function %rotl_i8_i32(i8, i32) -> i8 {
+block0(v0: i8, v1: i32):
+    v2 = rotl.i8 v0, v1
+    return v2
+}
+; run: %rotl_i8_i32(0xe0, 0) == 0xe0
+; run: %rotl_i8_i32(0xe0, 1) == 0xc1
+; run: %rotl_i8_i32(0xef, 0) == 0xef
+; run: %rotl_i8_i32(0xef, 4) == 0xfe
+; run: %rotl_i8_i32(0xe4, 64) == 0xe4
+; run: %rotl_i8_i32(0xe4, 65) == 0xc9
+; run: %rotl_i8_i32(0xe4, 66) == 0x93
+; run: %rotl_i8_i32(0xe4, 257) == 0xc9
+
+function %rotl_i8_i16(i8, i16) -> i8 {
+block0(v0: i8, v1: i16):
+    v2 = rotl.i8 v0, v1
+    return v2
+}
+; run: %rotl_i8_i16(0xe0, 0) == 0xe0
+; run: %rotl_i8_i16(0xe0, 1) == 0xc1
+; run: %rotl_i8_i16(0xef, 0) == 0xef
+; run: %rotl_i8_i16(0xef, 4) == 0xfe
+; run: %rotl_i8_i16(0xe4, 64) == 0xe4
+; run: %rotl_i8_i16(0xe4, 65) == 0xc9
+; run: %rotl_i8_i16(0xe4, 66) == 0x93
+; run: %rotl_i8_i16(0xe4, 257) == 0xc9
+
+function %rotl_i8_i8(i8, i8) -> i8 {
+block0(v0: i8, v1: i8):
+    v2 = rotl.i8 v0, v1
+    return v2
+}
+; run: %rotl_i8_i8(0xe0, 0) == 0xe0
+; run: %rotl_i8_i8(0xe0, 1) == 0xc1
+; run: %rotl_i8_i8(0xef, 0) == 0xef
+; run: %rotl_i8_i8(0xef, 4) == 0xfe
+; run: %rotl_i8_i8(0xe4, 64) == 0xe4
+; run: %rotl_i8_i8(0xe4, 65) == 0xc9
+; run: %rotl_i8_i8(0xe4, 66) == 0x93
+
+
+
+;; This is a regression test for rotates on x64
+;; See: https://github.com/bytecodealliance/wasmtime/pull/3610
```

Comment:
I really appreciate that you not only tracked down where these particular tests came from, but also left a link here in case anybody else wonders!